ts/Dashboards/Layout/Layout.ts

```diff
@@ -282,6 +285,12 @@ class Layout extends GUIElement {
     public destroy(): void {
         const layout = this;
 
+        for (let i = layout.board.layouts.length - 1; i >= 0; i--) {
+            if (layout.board.layouts[i] === layout) {
```

Comment:
Can we compare layouts by id? It will save us comparing the whole structure of layouts.

Reply:
When you are comparing the objects in JS it is done by checking the reference of the object, so IMO it is safer to check if this is the same object.